File: alumni/app/Models/Alumni.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Alumni extends Model
{
public function up()
{
    Schema::create('alumni', function (Blueprint $table) {
        $table->id('id_alumni');
        $table->string('nama');
        $table->text('alamat');
        $table->string('nis', 20);
        $table->string('nisn', 20);
        $table->string('password');
        $table->string('foto', 255)->nullable();
        $table->year('tahun_angkatan')->nullable();
        $table->string('nama_usaha', 255)->nullable();
        $table->string('nomor_telepon', 15)->nullable();
        $table->string('link_web_usaha', 255)->nullable();
        $table->string('email');
        $table->integer('id_kategori')->nullable();
        $table->string('role', 50)->default('alumni');
        $table->timestamps();
    });
}
}

File: lib/models/alumni.dart
class Alumni {
  final int id;
  final String nama;
  final String email;
  final String tahunAngkatan;
  final String alamat;
  final String nis;
  final String nisn;
  final String? namaUsaha;
  final String? nomorTelepon;
  final String? linkWebUsaha;

  Alumni({
    required this.id,
    required this.nama,
    required this.email,
    required this.tahunAngkatan,
    required this.alamat,
    required this.nis,
    required this.nisn,
    this.namaUsaha,
    this.nomorTelepon,
    this.linkWebUsaha,
  });

  // Fungsi untuk mengonversi JSON ke model Alumni
  factory Alumni.fromJson(Map<String, dynamic> json) {
    return Alumni(
      id: json['id'],
      nama: json['nama'],
      email: json['email'],
      tahunAngkatan: json['tahun_angkatan'].toString(),
      alamat: json['alamat'],
      nis: json['nis'],
      nisn: json['nisn'],
      namaUsaha: json['nama_usaha'],
      nomorTelepon: json['nomor_telepon'],
      linkWebUsaha: json['link_web_usaha'],
    );
  }

  // Fungsi untuk mengonversi model Alumni menjadi format JSON
  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'nama': nama,
      'email': email,
      'tahun_angkatan': tahunAngkatan,
      'alamat': alamat,
      'nis': nis,
      'nisn': nisn,
      'nama_usaha': namaUsaha,
      'nomor_telepon': nomorTelepon,
      'link_web_usaha': linkWebUsaha,
    };
  }
}
